src/gp_activetable.c

```diff
@@ -1173,21 +1125,26 @@ pull_active_table_size_from_seg(HTAB *local_table_stats_map, char *active_oid_ar
 		for (j = 0; j < PQntuples(pgresult); j++)
 		{
 			reloid    = atooid(PQgetvalue(pgresult, j, 0));
-			tableSize = (Size)atoll(PQgetvalue(pgresult, j, 1));
-			entry     = (ActiveTableEntryCombined *)hash_search(local_table_stats_map, &reloid, HASH_ENTER, &found);
+			tableSize = atoll(PQgetvalue(pgresult, j, 1));
+			segId     = atoi(PQgetvalue(pgresult, j, 2));
 
-			/* for diskquota extension version is 1.0, pgresult doesn't contain segid */
-			if (PQnfields(pgresult) == 3)
-			{
-				/* get the segid, tablesize for each table */
-				segId                       = atoi(PQgetvalue(pgresult, j, 2));
-				entry->tablesize[segId + 1] = tableSize;
-			}
+			update_active_table_size(reloid, tableSize, segId);
```

Comment:
I'll ask this question again:
Ar you sure that updating quotas here and original updating in calculate_table_sizes are equivalent?

Reply:
Why do you have doubts?

Reply:
> Do we skip them too?

yes, that tests passed

Reply:
tests are checking only table_size entries. Before current patch we couldn't reach quota update for those tables, now we call it

Reply:
>  because the invalidation is done later than the active table sizes are updated

give the line

Reply:
That problem was only during initialization and those tests showed it. I don't understand what

> quota update

has to do with it?

Reply:
> > because the invalidation is done later than the active table sizes are updated
> 
> give the line

```
refresh_disk_quota_usage
 +- gp_fetch_active_tables
   +- load_table_size
     +- update_active_table_size
 +- calculate_table_disk_usage
   +- invalidation when is_init
```

Reply:
>That problem was only during initialization and those tests showed it

I am talking not about your case specifically, I meant the original [code](https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L993) where `continue` takes place.

Why it is ok to give up on this `continue` thing in our case?

Reply:
> Why it is ok to give up on this `continue` thing in our case?

Active tables are always in `pg_class`, which means a branch https://github.com/arenadata/diskquota/blob/767d498fd60664cb27ab2d216521d26ed0a4cb83/src/quotamodel.c#L957-L968 is executed for them without a patch.

Reply:
And during initialization , when we are scanning table_size, not always?

Reply:
During initialization, if the table was already deleted, but we read it from `table_size`, then the patch updates the sizes for it, and then during invalidation it deletes it from the shared hashmap.

Reply:
That what i wanted to hear

Reply:
Although, perhaps we should add additional conditions to the `update_active_table_size` function so that it first checks the table in `pg_class` and `relation_cache`, as is done in the `calculate_table_disk_usage` function.

Reply:
In this case sys cache searching will be done extra times ( in  update_active_table_size and in calculate_table_disk_usage instead of calculate_table_disk_usage only)

Reply:
Unfortunately yes, and `relation_cache` too. Suggestions?

Reply:
>Although, perhaps we should add additional conditions to the update_active_table_size function so that it first checks the table in pg_class and relation_cache, as is done in the calculate_table_disk_usage function.

How critial it will be if we dont't filter such oids?